turbopack/crates/turbo-tasks/src/vc/resolved.rs

```diff
@@ -221,6 +221,21 @@ where
         }
     }
 
+    /// Upcasts the given `Vec<ResolvedVc<T>>` to a `Vec<ResolvedVc<K>>`.
+    ///
+    /// See also: [`Vc::upcast`].
+    #[inline(always)]
+    pub fn upcast_vec<K>(vec: Vec<Self>) -> Vec<ResolvedVc<K>>
+    where
+        T: UpcastStrict<K>,
+        K: VcValueTrait + ?Sized,
+    {
+        debug_assert!(size_of::<ResolvedVc<T>>() == size_of::<ResolvedVc<K>>());
+        debug_assert!(size_of::<Vec<ResolvedVc<T>>>() == size_of::<Vec<ResolvedVc<K>>>());
+        // Safety: The memory layout of `ResolvedVc<T>` and `ResolvedVc<K>` is the same.
+        unsafe { transmute::<Vec<ResolvedVc<T>>, Vec<ResolvedVc<K>>>(vec) }
```

Comment:
I agree with transmuting `ResolvedVc` being safe as it's basically a `#[repr(transparent)]` of a `RawVc`, but transmutes of `Vec` should always use `Vec::from_raw_parts`: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_raw_parts